napytau/gui/components/graph.py

```diff
@@ -73,12 +81,62 @@ def plot(self, value: int, appearance: str) -> Canvas:
             True, which="both", color=secondary_color, linestyle="--", linewidth=0.3
         )
 
-        # plotting the graph
-        plot1.plot(y)
+        # plotting the datapoints with appropriate markers
+        for index in range(len(y_data)):
+            # Generate marker and compute dynamic markersize
+            marker = MarkerStyle(generate_marker(distances[index]))
+            size = distances[index] * 5  # Scale markersize based on distance
+            plot1.plot(
+                x_data[index],
+                y_data[index],
+                marker=marker,
+                linestyle="None",
+                markersize=size,
+                label=f"Point {index + 1}",
+                color=main_marker_color,
+            )
+
+        # plotting fitting curve of datapoints
+
+        coeffs = np.polyfit(x_data, y_data, 10)  # Calculating coefficients
+
+        poly = np.poly1d(coeffs)  # Creating polynomial with given coefficients
+
+        x_fit = np.linspace(min(x_data), max(x_data), 100)
+        y_fit = poly(x_fit)
+
+        plot1.plot(
+            x_fit, y_fit, color="red", linestyle="--", linewidth="0.6"
+        )  # plotting
 
         # creating the Tkinter canvas
         # containing the Matplotlib figure
         canvas = FigureCanvasTkAgg(fig, master=self.parent)
         canvas.draw()
 
         return canvas.get_tk_widget()
+
+
+# Generate custom marker function
```

Comment:
This confuses me more than it explains anything. Also consider using the doc comment style when commenting on a function
```suggestion
def generate_path(distance: float) -> Path:
"""Create a path to describe how an error marker should be drawn around a data point"""
```